File: evidence/generated-systems/golang/analytics-on-golang/analytics-on-golang/internal/model/DataSet.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// DataSet Declaration
//==============================================================
type DataSet struct {
    gorm.Model
     Name                                    string
    SchemaVersion                                    string
    RefreshSchedule                                                            string
    Sensitive                                    bool
    WorkspaceId         *uint
    Workspace           *AnalyticsWorkspace `gorm:"foreignKey:WorkspaceId"`
     Sources           []DataSource `gorm:"foreignKey:SourcesFromDataSetId"`
     Pipelines           []DataPipeline `gorm:"foreignKey:PipelinesFromDataSetId"`
     SemanticModels           []SemanticModel `gorm:"foreignKey:SemanticModelsFromDataSetId"`
     Dimensions           []Dimension `gorm:"foreignKey:DimensionsFromDataSetId"`
     Measures           []Measure `gorm:"foreignKey:MeasuresFromDataSetId"`
     Metrics           []Metric `gorm:"foreignKey:MetricsFromDataSetId"`
     QualityRules           []QualityRule `gorm:"foreignKey:QualityRulesFromDataSetId"`
    LineageNodeId         *uint
    LineageNode           *LineageNode `gorm:"foreignKey:LineageNodeId"`
     Tags           []Tag `gorm:"foreignKey:TagsFromDataSetId"`
    DataFormat                      DataFormat

// parent associations as their child

}

File: evidence/generated-systems/rubyonrails/analytics-on-rails/analyticsonrails/app/models/dataset.rb
class DataSet < ApplicationRecord
  enum DataFormat: [:CSV, :JSON, :Parquet, :Avro, :ORC, :XML]


  composed_of :cronSchedule,
    class_name: "CronSchedule",
    mapping: [
      ${$mapping}, 
      %w[cronSchedule_timezone timezone]
    ]

  has_many :Workspace, class_name: 'AnalyticsWorkspace'
  has_many :Sources, class_name: 'DataSource'
  has_many :Pipelines, class_name: 'DataPipeline'
  has_many :SemanticModels, class_name: 'SemanticModel'
  has_many :Dimensions, class_name: 'Dimension'
  has_many :Measures, class_name: 'Measure'
  has_many :Metrics, class_name: 'Metric'
  has_many :QualityRules, class_name: 'QualityRule'
  has_many :LineageNode, class_name: 'LineageNode'
  has_many :Tags, class_name: 'Tag'

end
